processing/src/main/java/io/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -63,6 +59,20 @@
   public abstract Object combine(Object lhs, Object rhs);
 
   /**
+   * Creates an AggregateCombiner to combine rollup aggregation results from serveral "rows" of different indexes during
+   * index merging. AggregateCombiner implements the same logic as {@link #combine}, with the difference that it uses
+   * {@link io.druid.segment.ColumnValueSelector} and it's subinterfaces to get inputs and implements {@code
+   * ColumnValueSelector} to provide output.
+   *
+   * @see AggregateCombiner
+   * @see io.druid.segment.IndexMerger
+   */
+  public AggregateCombiner makeAggregateCombiner()
+  {
+    throw new UOE("[%s] does not implement makeAggregateCombiner()", this.getClass().getName());
```

Comment:
This is still incompatible, because aggregators that could have been used during ingestion will not be able to be used after this. The aggregators in question arguably only make sense at query time, but it is still an incompatibility.